modules/ROOT/pages/header-inject-remove-task.adoc

```diff
@@ -1,54 +1,70 @@
-= Header Injection and Removal
+= Header Injection
 ifndef::env-site,env-github[]
 include::_attributes.adoc[]
 endif::[]
 
 [width="100%", cols="5,15"]
 |===
->s| Policy Name | Header injection/Header removal
->s|Summary      | Adds/removes headers to a request or a response
+>s| Policy Name | Header Injection
+>s|Summary      | Adds headers to a request or a response
 >s|Category | Transformation
 >s| First Mule version available | 4.1.0
 .1+>.^s| Returned Status Codes
 |
 |===
+
+//any ret codes to add here Mariano?
     
-The Header Injection policy adds headers to the request/response of the message. The Header Removal policy removes all listed headers.
+The Header Injection policy adds headers to the request or response of the message. When you configure this policy for your API, you must specify an inbound and outbound map of the headers that you want to add in the message processing in the form of a key-value pair. 
 
 image::header-inject-remove-diagram.png[]
 
+You can optionally include Mule runtime engine (Mule) expressions for Mule 3.0 and DataWeave expressions for Mule 4.0 in the value of the header:
+
+`“#[attributes.requestPath]”`
 
-== Applying Header Injection Policy
 
-To apply this policy to an API, specify an inbound/outbound map of the headers that you want to add in the message processing in the form of a key-value pair.
 
-image::inbound-header-map.png[]
+== How This Policy Works
 
-image::outbound-header-map.png[]
+If the injected header already exists in the message attributes, the policy creates a new header without overriding the value of the existing one. This action creates two headers with the same name. To overwrite the value of a header, you must remove it before injecting the header.  
```

Comment:
Shouldn't we reiterate that the policy will add the desired header in the specified pointcut (request or response)? Then continue with what happens if the header is already present?

Reply:
Also, do we want to point out to the removal policy to guide the user on how to remove the header first?

Reply:
Done! Thanks!